ecosystem/sep-0024.md

```diff
@@ -190,6 +192,127 @@ The deposit flow can only be fulfilled if the Stellar `Account` has established
+**Anchors and Wallets**: Both anchors and wallets must also support the aforementioned ["stellar account doen't have a trust line for that asset."](#stellar-account-doesnt-trust-asset).
+Wallets need to be interoperable with anchors that have not adopted the [claimable balance feature](#Claimable-Balances) and vice versa.
+
+##### Wallet Claimable Balance Flow
+
+Wallets will still follow the [typical deposit flow](#Basic-Wallet-Implementation).
+Wallets will provide `claimable_balance_supported=true` field and value in their POST parameters for `/transactions/deposit/interactive`.
+
+If the anchor we're depositing to **supports** claimable balances.
+
+1. Make a request to `/transactions/deposit/interactive` and provide a value for the `claimable_balance_supported=true` field.
+1. When the transaction status becomes `pending_user_transfer_start` send the required payment as described in the interactive webapp callback or the `/transaction` endpoint.
+1. The anchor will report transaction status as `pending_anchor` then make the payment as a claimable balance operation.
+1. After transaction is submitted the anchor will report transaction status as `completed`
+
+If the anchor we're depositing to **doesn't support** claimable balances.
+
+1. Make a request to `/transactions/deposit/interactive` and provide a value for the `claimable_balance_supported=true` field.
+1. When the transaction status becomes `pending_user_transfer_start` send the required payment as described in the interactive webapp callback or the `/transaction` endpoint.
+1. The anchor will ignore the `claimable_balance_supported` field, discover the account does not have a trust line, and report transaction status as `pending_trust`.
+1. At this point both the wallet and anchor are waiting on the trust line to be established in order to continue the deposit payment. This is described here [steps 4-6](#stellar-account-doesnt-trust-asset)
+
+##### Wallet Feature: Claimable Balance Polling
+
+Claimable Balance Polling allows the wallet to update the claimant that their is a balance available to claim and they may need to establish a [trust line](#stellar-account-doesnt-trust-asset) in order to do so.
+There are a few options as to how a wallet can poll and surface claimable balance information for their users.
+
+1. The wallet can poll Horizon's [`/claimable_balances/`](https://developers.stellar.org/api/resources/claimablebalances/list/) list endpoint to pull any available claimable balances for the claimant.
+
+   * *Recommended request* `GET /claimable_balances?claimant={:claimant}&order={asc,desc}&limit={1-200}` or [See List Claimable Balances](https://developers.stellar.org/api/resources/claimablebalances/list/)
+
+2. The wallet can poll Horizon's [`/claimable_balances/`](https://developers.stellar.org/api/resources/claimablebalances/single/) single endpoint to pull the submitted transactions' available claimable balance(s) for the claimant(user) using the claimable_balance_id.
```

Comment:
I don't think this option will be valuable for wallets, because they won't know the balance ID until they stream the transaction or poll the anchor for the `claimable_balance_id`. In either scenario, they won't need to make an additional Horizon request to get info on the balance, they can just immediately submit the claim op.

Not to mention that decoding the result XDR and getting the balance ID is complex, as evident by how much you had to write in order to explain it. I vote to remove this method.